### PR TITLE
Integrate react-navigation

### LINKED ISSUE #4866: Integrate native navigation transitions
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a User, I want to have native transitions while navigating between views so that I have a clear understanding of my location inside the app, how I got here and where I can return to. 

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: iOS provides a consistent visual framework of transitions between views. I'd like to suggest we find a way to support those native behaviours, to my basic knowledge integrating **React Navigation** would allow just that. 

#### Expected behavior
Tapping the Back Button in an opened Chat View would smoothly transition the view while placing the previous Home view in its place. The animated segue transition should work both ways: tapping on a chat table cell should smoothly present the detailed chat view above it. 

The same logic should follow other views that are presented onto a navigation stack like choosing a currency from settings, viewing help, tapping cells with disclosure indicators: sending/ receiving transactions, displaying transaction history. 

#### Actual behavior
Views are presented without any transition when navigating between them.

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: 0.9.21 (6078)
* Operating System: iOS

#### Logs

[comment]: # (if on Android please replicate bug whilst running adb logcat)
```
...
```

### PR DESCRIPTION
**Pull request by rasom**
Fixes #4866

status: ready

### CODE REVIEW

**Comment by chadyj** [comment]
Nice one! Its a joy to swipe back. Feels much more native and fluid.
Some quick WIP feedback: iPhone X is completely messed up with actions in the safe areas (eg top left, top right, bottom).

**Comment by chadyj** [comment]
Great progress!  

I'm sure it is on your list but just in case I ran into these issues:

`Wallet > Collectibles`
`Dapp > Wallet modal` 
`public chat > tap user > view profile` 
`Wallet > Manage assets`

**Comment by vitvly** [comment]
Cool PR! Just wanted to report that I tried running the branch on Desktop, and `:navigate-to :create-account` invoked on the initial screen does nothing. Only had a brief glance at the PR changes, so cannot really say what is wrong there.

**Comment by rasom** [comment]
@siphiuel , it's broken because `:navigate-to` and other navigation related handlers are changed. I will first make sure it works on mobile, then figure how to keep existing logic for the desktop.

**Comment by rasom** [comment]
known issues on iphone X
1. status test dapp -> deploy simple contract -> x
2. logout
3. status test dapp -> send one tx -> advanced -> transaction fee -> done
4. on sending transaction/signing screens password input is hidden by keyboard
5. "next" button is hidden by keyboar in "create account" flow

**Comment by rasom** [comment]
@siphiuel desktsop version should be working now

File: env/dev/env/ios/main.cljs [hunk 1]
```diff
@@ -20,7 +20,9 @@
 
 (figwheel/start {:websocket-url    (:ios conf/figwheel-urls)
                  :heads-up-display false
-                 :jsload-callback  #(swap! cnt inc)})
+                 :jsload-callback  (fn []
+                                       (println :CHANGES_LOADED)
```
Comment (cammellos) [inline env/dev/env/ios/main.cljs]: debug statement?

Comment (rasom) [reply env/dev/env/ios/main.cljs]: yeah, i will remove it though i found it really useful sometimes

Comment (mandrigin) [reply env/dev/env/ios/main.cljs]: maybe make a `(log/debug)` entry then?

File: src/status_im/ui/screens/wallet/transaction_fee/views.cljs [hunk 2]
```diff
@@ -19,7 +19,8 @@
 (defn return-to-transaction [modal?]
   (if modal?
     ;;TODO(andrey) artificial navigation stack for modals (should be reworked)
-    (re-frame/dispatch [:navigate-to-modal :wallet-send-transaction-modal])
+    ;; ^ probably will be fixed with react-navigation
```
Comment (mandrigin) [inline src/status_im/ui/screens/wallet/transaction_fee/views.cljs:22]: is it fixed?

**Review by mandrigin — APPROVED** [review]
quite a lot of changes, but look reasonable

**Comment by denis-sharypin** [comment]
1. Background color should be white here. Seems like some regression


2. When you are opening a dapp it redirects you to a home screen for one second. Doesn't look right. 
Can we open a dapp right directly from the dapp profile list? And then when you close it you will get the home screen?

3. The same is when you are adding somebody via contact code it redirects you to a home screen for a sec. Also, there is some misplacement of the input. 


4. Wrong background for the whole backup transaction flow


5. A bit weird transaction when you are setting up wallet. It gets backward after you are confirming a Transaction phrase

@asemiankevich  could you please test out send transactions flows?

File: src/status_im/chat/commands/impl/transactions.cljs [hunk 3]
```diff
@@ -262,31 +263,37 @@
...
+                 (transaction-details recipient-contact symbol))
+                (update-in [:wallet :send-transaction]
+                           dissoc :id :password :wrong-password?))
+        ;; TODO(janherich) - refactor wallet send events, updating gas price
+        ;; is generic thing which shouldn't be defined in wallet.send, then
+        ;; we can include the utility helper without running into circ-dep problem
+        :update-gas-price {:web3          (:web3 db)
+                           :success-event :wallet/update-gas-price-success
+                           :edit?         false})
+       (navigation/navigate-to-cofx next-view-id {}))))
```
Comment (janherich) [inline src/status_im/chat/commands/impl/transactions.cljs]: This is strange, `merge-fx` should be used in this style when fixed map of effects in addition to cofx functions is required:
`(merge-fx cofx {:db modified-db :another-effect another-effect} (cofx-producing-function))`
Why did you change it to modify the first parameter (coeffects map) itself ?

File: src/status_im/ui/screens/navigation.cljs [hunk 4]
```diff
@@ -51,20 +50,24 @@
     (apply preload-data! db args)))
 
 (defn navigate-to-cofx [go-to-view-id screen-params {:keys [db]}]
```
Comment (janherich) [inline src/status_im/ui/screens/navigation.cljs]: Why it's named `navigate-to-cofx` instead of just `navigate-to` ? Is there some reason why `navigate-to` function should be exposed as public, opposed to make it just `internal-navigate-to` private helper function in the navigation ns ?

Comment (cammellos) [reply src/status_im/ui/screens/navigation.cljs]: For the naming https://github.com/status-im/status-react/blob/f36d5d64cc52ccd6bb06fb8743ae9ed535b21d0d/src/status_im/ui/screens/navigation.cljs#L62 
`navigate-to` was already taken and wasn't playing nice with `merge-fx`, so I  started using the `cofx` version leaving `navigate-to` to use the `cofx` version, so we can slowly move away from it.

Comment (rasom) [reply src/status_im/ui/screens/navigation.cljs]: anyways a bit out of the scope of this PR, we can return to this naming later

File: src/status_im/chat/commands/impl/transactions.cljs (hunk 3 above)
Comment (rasom) [reply src/status_im/chat/commands/impl/transactions.cljs]: fixed

File: src/status_im/ui/screens/navigation.cljs (hunk 4 above)
Comment (janherich) [reply src/status_im/ui/screens/navigation.cljs]: I agree, let's settle the naming later

**Review by janherich — APPROVED** [review]

**Comment by churik** [comment]
@dmitryn thank you, now I can see both popups (Jenkins mystery).
I've found one more small issue - I can see `Request 400, STT` in chat.

*Steps:*
1) open 1-1 chat
2) type /request STT 20, 0005
3) send message

*Expected result:*
`/request 20 STT` 
*Actual result:*
`/request 20, STT`
req
Issue is reproducible for /send and /request commands in chat with . and , in the end
Would you mind to fix it?

**Comment by rasom** [comment]
@denis-sharypin please check latest build

**Comment by churik** [comment]
@rasom 
1, 2, 3, 4 are fixed;

### 5) Some screens still have grey background

- Network details


- Add network
add

- Add bootnode
add_b

-  Inside collectibles page
image


### 6) Error at attempt to send ETH in 1-1 chat if you edit Transaction Fee details
```
'Error: Error: Error: Error: There is no route defined for key wallet-transaction-sent. Must be one of: 'wallet-send-transaction-modal','wallet-transaction-sent-modal'
```
*Steps:*

1) use /send command in 1-1 chat
2) on Send transaction screen tap on Advanced
3) change Gas Price and tap on Done
4) sign transaction

'Error: Error: Error: Error: There is no route defined for key wallet-transaction-sent. Must be one of: 'wallet-send-transaction-modal','wallet-transaction-sent-modal'
TF session: https://app.testfairy.com/projects/4803590-status/builds/8581810/sessions/4400738760/?accessToken=7TNjRzSbMydJxSNrOCVBaPMlmZ4

### 7) Input text field is misplaced for few moments when navigating using "Swipe" from user profile

*Steps:*
- open public chat
- tap on any user pic
- swipe to come back

image

*TF session:* https://app.testfairy.com/projects/4803590-status/builds/8581810/sessions/4400629969/?accessToken=4CEv511iT13Gk8OY6JEEGRTNna4
Same when you swipe from user profile to go back to 1-1 chat.

### 8) Got an error at attempt to view collectibles

*Steps:*
- navigate to wallet
- open collectibles (cryptostickers in my case)
- tap on "View"
```
NAME: "Error"
MESSAGE: Error: Error: Error: Error: There is no route defined for key home.
Must be one of: 'wallet-onboarding-setup','wallet-request-transaction','contact-code','wallet-send-transaction-request','wallet-send-transaction-chat','wallet','recipient-qr-code','wallet-send-transaction','collectibles-list','recent-recipients','wallet-send-assets','wallet-request-assets','wallet-transaction-sent','transactions-history','unsigned-transactions','wallet-transaction-details'
```
testfairy status 0 9 28 7643 2018 2018-09-05 18-31-33

*TF session (06:40 - 06:50):*  https://app.testfairy.com/projects/4803590-status/builds/8581810/sessions/4400630242/?accessToken=Me2IQe9wPeSVITYZaaeCPAGaEMw

### 9) Keyboard overlaps "Done" and "Reset to default", so impossible to edit Transaction fee

*Steps:*
- navigate to wallet
- tap on Send
- tap on Advanced

dsfdf

**Comment by rasom** [comment]
@churik i can't reproduce 6) in the last build

**Comment by churik** [comment]
@rasom thanks! 
Investigated 6 issue - steps are added, TF is updated.
Added 9 issue.
Can you please also resolve conflicts and rebase it to latest develop to check desktop build?

**Comment by churik** [comment]
Tested on [MacOS](https://status-im.ams3.digitaloceanspaces.com/StatusIm-180906-125321-5d2772-pr.dmg):
- installation
- creating account
- restoring account
- general navigation

**Comment by asemiankevich** [comment]
@rasom it is super weird but no account is stored when i change network on iOS. 

Steps:
- install status on iOS (iphone 7, 11.4)
- create account 
- change network to ropsten
- wait until app relogins u

Expected: you are on the login screen to enter password for the account you just created

Actual: you are at the very first screen of the app prompting you to create account or to restore it. No previous account is shown

NOTE: it appears when i restart the app. Seems wrong screen is shown upon network switching

**Comment by rasom** [comment]
ok, i'll fix this asap

**Comment by asemiankevich** [comment]
1) `Share my contact code` screen does not fit screen size (Huawei P10 Lite, screen size 5.2, Android 8)

image

2) when sending ETH from wallet to contact an error shown

Steps:
 - open status
- add a contact
- faucet some funds
- go to wallet and send some ETH to a contact

```
09-10 01:47:47.309  4826  4886 I ReactNativeJS: NAME: "Error"
09-10 01:47:47.309  4826  4886 I ReactNativeJS: MESSAGE: Error: Error: Error: Error: There is no route defined for key wallet-transaction-sent-modal.
09-10 01:47:47.309  4826  4886 I ReactNativeJS: Must be one of: 'wallet-onboarding-setup','wallet-request-transaction','contact-code','wallet-send-transaction-request','wallet-send-transaction-chat','wallet','recipient-qr-code','wallet-send-transaction','collectibles-list','recent-recipients','wallet-send-assets','wallet-request-assets','wallet-transaction-sent','transactions-history','unsigned-transactions','wallet-transaction-details'
09-10 01:47:47.309  4826  4886 I ReactNativeJS: 
09-10 01:47:47.309  4826  4886 I ReactNativeJS: This error is located at:
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in r
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in RCTView
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in RCTView
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in RCTView
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in RCTView
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in r
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in n
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in reagent2
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in root
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in reagent1
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in RCTView
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in RCTView
09-10 01:47:47.309  4826  4886 I ReactNativeJS:     in t
```

image

**Comment by asemiankevich** [comment]
@rasom this issue with iOS still happens upon logout (looks good for network switching).

**Comment by rasom** [comment]
i didn't fix anything, just added logging :D

**Comment by rasom** [comment]
@asemiankevich all previous bugs should be fixed in the latest commit

**Comment by asemiankevich** [comment]
@rasom i will check it tomorrow more precisely, but for now i noticed that when i tap `Advanced` on Profile screen - the screen is not autoscrolled. 

- Steps: Tap Profile -> Tap Advanced

**NOTE:** works fine on Wallet screen

Also, when i open  and close [SHARE MY CONTACT CODE] qr screen - small `no value` title is shown at the top left for a second. I could not make a screenshot but here are steps:

- Open profile -> tap Share my contact code -> tap cross button at the top left (or DONE on iOS)

Manage Assets screen does not fit screen size ( 5.2") on Android

image

**Comment by asemiankevich** [comment]
Blank screen when sending transaction in 1x1 chat on iOS (iPhone 7, 11.4)
Steps:
- open status
- open 1x1 chat with someone on iOS
- send ETH / token 



https://app.testfairy.com/projects/4803590-status/builds/8592333/sessions/4401463032/?accessToken=Sjcg5T8oH/4AU7kP9NSLDJDQyjI

When i got the blank screen i restarted the app and got this

**Comment by rasom** [comment]
1. should be fixed
2. can't reproduce / not sure this is something specific to this branch

**Comment by churik** [comment]
@rasom 
In Status-desktop Back arrow is not clickable on "Sign In", "Sign In to another account" screens
back_b
**Video:** http://take.ms/prXYa
Issue is not reproducible in nightly 14/09/2018, so appeared in PR.

<blockquote><div>Monosnap screenshot tool</div><div><strong><a href="https://monosnap.com/file/jVpfJVN0y0ONkoENkE0PtfVSkFGVid">File "screencast 2018-09-14 12-22-14.mp4"</a></strong></div><div>Monosnap — the best tool to take & share your screenshots.</div></blockquote>

**Comment by rasom** [comment]
@churik is that the only place in app were navigation back doesn't work properly? can you check other cases as well? So any "navigate back" arrow/button counts :)

**Comment by rasom** [comment]
also i would prefer to merge with a few small bugs and solve them asap in other PRs, because merging this change becoming a bit exhausting. Different bugs pop up because of refactoring/changes from other PRs

**Comment by churik** [comment]
@rasom I navigated through the whole app and didn't notice another issues

File: components/src/status_im/ui/components/react.cljs [hunk 5]
```diff
@@ -46,7 +46,13 @@
 
 (def text-class (get-class "Text"))
 (def text-input-class (get-class "TextInput"))
-(def image (get-class "Image"))
+(def image-class (get-class "Image"))
+(defn image [{:keys [source] :as props}]
+  (when (or (not (map? source))
+            (not (contains? source :uri))
```
Comment (yenda) [inline components/src/status_im/ui/components/react.cljs]: Could this be factored into a 'valid-source?' function ?

**Comment by rasom** [comment]
ok, thanks @churik

File: deps.edn [hunk 6]
```diff
@@ -12,7 +12,8 @@
          hickory                     {:mvn/version "0.7.1"}
          com.cognitect/transit-cljs  {:mvn/version "0.8.248"}
          status-im/pluto             {:mvn/version "iteration-2-SNAPSHOT"}
-         mvxcvi/alphabase            {:mvn/version "1.0.0"}}
+         mvxcvi/alphabase            {:mvn/version "1.0.0"}
+         rasom/cljs-react-navigation {:mvn/version "0.1.4"}}
```
Comment (yenda) [inline deps.edn:16]: Can you push to status repos ?

File: env/dev/env/ios/main.cljs (hunk 1 above)
Comment (yenda) [reply env/dev/env/ios/main.cljs]: Don't forget to remove it or turn it into a debug statement

File: project.clj [hunk 7]
```diff
@@ -12,7 +12,8 @@
                  [hickory "0.7.1"]
                  [com.cognitect/transit-cljs "0.8.248"]
                  [status-im/pluto "iteration-2-SNAPSHOT"]
-                 [mvxcvi/alphabase "1.0.0"]]
+                 [mvxcvi/alphabase "1.0.0"]
+                 [rasom/cljs-react-navigation "0.1.4"]]
```
Comment (yenda) [inline project.clj:16]: Move to status repo

**Review by yenda — APPROVED** [review]
Some of the names are confusing or not informative enough but I agree that it's better to merge for now and clean up later. Kudos for the hard work!

File: deps.edn (hunk 6 above)
Comment (rasom) [reply deps.edn:16]: yes sure

File: project.clj (hunk 7 above)
Comment (rasom) [reply project.clj:16]: though this is about uberjar... let's fix this in a separate PR

File: components/src/status_im/ui/components/react.cljs (hunk 5 above)
Comment (rasom) [reply components/src/status_im/ui/components/react.cljs]: done

**Comment by rasom** [comment]
@churik please check navigation back on these screens

**Comment by churik** [comment]
@rasom 
new issue appeared - when you navigate from to "Create new account" and then Back - blank screen is shown.

*Video:* http://take.ms/aoC1J

<blockquote><div>Monosnap screenshot tool</div><div><strong><a href="https://monosnap.com/file/iaT3yEhy6Dk4I3NOemoDFN28NTwkrP">File "screencast 2018-09-14 14-57-33.mp4"</a></strong></div><div>Monosnap — the best tool to take & share your screenshots.</div></blockquote>

**Comment by asemiankevich** [comment]
So the issues i see now:
- advanced does not scroll down the Profile screen
- navigation is not  smooth , as well as QR code appearance is too
- gas price is not fetched but this could be due poor connection i have now
- application closes when recovering account or  trying to login to recovered account on ios build https://i.diawi.com/UYFbn1 . So i cant login to recovered account on ios

<blockquote><div><strong><a href="https://i.diawi.com/UYFbn1">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by Serhy** [comment]
Latest build (https://status-im.ams3.digitaloceanspaces.com/StatusIm-180914-104610-0d56f7-pr.apk) issues for Android (Android 7, 6.0.1 confirmed):

1) 'Send transaction' screen is opened when I send 'New request' from Wallet after 'Choose Recipient' is selected/declined. To reproduce:
- Open Status and log in to account
- Navigate to `Wallet` -> `Receive transaction` -> `Send tranaction request`
- Tap `Choose recipient` 
- Close this screen by top-left `<` icon
**Actual result:** Send tranaction screen has opened

**Note:** if to select a recipient from this list `Send transaction` screen will be there as well. So `Specify recipient` requests `Send transaction`

TF session: https://app.testfairy.com/projects/4803622-status/builds/8593208/sessions/4401494209/?accessToken=2yBOz6R/1Q-IGZ-Vwp5lU7Lq86Y

2) Some of the screens (apparently those which opens in a modal window) do not consider the **device** status bar in this PR version, so Status app top bar on such screens got overlapped by device status bar. Affected screen views are:
- Open Status, log in to account
- Navigate to any **public chat** and open **modal wallet screen** by the top wallet icon 
- From there tap open `<-/->` top-right button to swich on to the tranaction history screen
- Navigate to any **Dapp** and open **modal wallet screen** by the top wallet icon 
- Navigate to main Wallet screen -> three dot at top-right -> **Manage assets** menu

TF session: https://app.testfairy.com/projects/4803622-status/builds/8593208/sessions/4401496553/?accessToken=rlmSBBnvbuGJFvTY9c5LNz0kyGQ

3) Android device 'Back' button doesn't bring any effect for Wallet screens which has been opened from Dapp. **To reproduce:**
- Open Status and log in to account. Enable Developer mode from the profile screen 
- Navigate to `Open Dapp` -> `Status Test Dapp` -> `Open`
- Switch to Transactions tab and tap `Deploy simple contract`
- Open `Transaction fee ` screen 
- Tap device `Back` button
**Actual result:** Nothing happens
**Expected result:** Transaction fee screen closed and contact returned back to Send transaction screen with the default values applied as Gas limit/Gas price

- Close Transaction Fee screen by the top-left cross icon
- Now you should be on 'Send transaction'. Tap device `Back` button again.
**Actual result:** Nothing happens
**Expected result:** `Send transaction` screen closed and contact returned back to Dapp page from where Wallet screen been requested

TF session: https://app.testfairy.com/projects/4803622-status/builds/8593208/sessions/4401496751/?accessToken=VZIKeSnQEn1WJVM9xNYBxao4Iz8

4) Same as above 3rd item but for Transaction Fee screen opened from `Wallet` -> `Send request`
- Open Status and log in to account
- Navigate to `Wallet` -> `Send transaction` -> `Advanced` -> `Transaction Fee`
- Tap Android device `Back` button
**Actual result:** No effect
**Expected result:** Transaction fee screen closed and contact returned back to Send transaction screen with the default values applied as Gas limit/Gas price


<blockquote></blockquote>
<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8593208/sessions/4401494209/?accessToken=2yBOz6R/1Q-IGZ-Vwp5lU7Lq86Y">TestFairy: Status 0.9.28 (8258) @2018</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8593208/sessions/4401496553/?accessToken=rlmSBBnvbuGJFvTY9c5LNz0kyGQ">TestFairy: Status 0.9.28 (8258) @2018</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8593208/sessions/4401496751/?accessToken=VZIKeSnQEn1WJVM9xNYBxao4Iz8">TestFairy: Status 0.9.28 (8258) @2018</a></strong></div></blockquote>

**Comment by rasom** [comment]
> application closes when recovering account or trying to login to recovered account on ios build https://i.diawi.com/UYFbn1 . So i cant login to recovered account on ios

This one got fixed without me touching anything. I wasn't able to reproduce it locally, now it just works. Mystery. I suppose that's some jenkins related issue.

<blockquote><div><strong><a href="https://i.diawi.com/UYFbn1">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@rasom one critical issue I noticed so far is that I am logged out when I press Android back button from the Wallet tab. The app closes and when I open it again and try to log in ther eis `Unknown status-go error` tooltip. I can't find TF session for it. 

Everything works fine when back button is pressed in Home tab. The app closes and I'm still logged in when I open it again.

I'm testing on Galaxy S8, Android 8.0.

**Comment by rasom** [comment]
@lukaszfryc, ok, im on it

**Comment by lukaszfryc** [comment]
FYI, @Serhy's [issues](https://github.com/status-im/status-react/pull/5606#issuecomment-421360444) are fixed 👍 There are few minor things like transaction fee screen not being expanded all the way to the bottom of the screen, but, I will report them as separate issues.

**Comment by lukaszfryc** [comment]
Also, things Nastya [reported](https://github.com/status-im/status-react/pull/5606#issuecomment-421340389) also work fine now. Apart from navigation not beeing smooth and gas issue which is not related (I also seen it in other PRs).

**Comment by lukaszfryc** [comment]
I also had `null is not an object, t.minus()` error while sending ETH but I think it's in develop as well since I was able to repro it in [other PR](https://github.com/status-im/status-react/pull/5817#issuecomment-421933839). Here is this issue https://github.com/status-im/status-react/issues/5392

**Comment by Serhy** [comment]
iPhoneX specific issue:
When Wallet screen is called from 1-1 chat /send request or Dapp transaction - there are several overlapped cases are present (**Note**: it's all fine when Send transaction right from `Wallet` -> `Send tranaction` screen, so on the below screenshots: 
**right side screenshot is the issue** i.e. what's happening when Wallet opened from`1-1 chat -> /send ETH 0` OR `Dapp -> Sign tranaction` and **left side screenshot is expected state** (taken from `Wallet` -> `Send tranaction`)  :
1) `Sign transaction` button jumped up a bit
screen shot 2018-09-17 at 15 37 36


2) `Cancel +  Sign tranaction` buttons overlapped by password pop-up when signing tranaction:
screen shot 2018-09-17 at 15 11 51

3) On Tranaction Fee screen when cursor is put in editbox (gas Price/ gas Limit) keyboard the `Reset to default .... Done` area is too close to the numeric keyboard.

screen shot 2018-09-17 at 15 10 49

**Comment by lukaszfryc** [comment]
I also created a separate issue (can be fixed after this PR) about https://github.com/status-im/status-react/issues/5876. It maybe related to your [comment](https://github.com/status-im/status-react/pull/5606#issuecomment-421998018) @Serhy.

**Comment by lukaszfryc** [comment]
@rasom @Serhy re Desktop app, I think we are good to go 👍

> In Status-desktop Back arrow is not clickable on "Sign In", "Sign In to another account" screens

Is fixed 👍 

> new issue appeared - when you navigate from to "Create new account" and then Back - blank screen is shown. Video: http://take.ms/aoC1J

This issue still occurs but I think we can fix it later. I created https://github.com/status-im/status-react/issues/5877

**Comment by rasom** [comment]
`2.` `3.` are fixed

**Comment by lukaszfryc** [comment]
> when I press Android back button from the Wallet tab. The app closes and when I open it again and try to log in ther eis `Unknown status-go error` tooltip. I can't find TF session for it.

This issue is fixed 👍 

----

But @rasom, I noticed that only on Android (I tested on S8, Android 8.0), the transaction fee screen shows buttons from the previous screen (sign transaction). See:



It's probably related to #5876

Steps to reproduce:
- Switch to Ropsten and open Test Dapp
- Deploy simple contract
- Click advanced button and go to transaction fees

**Comment by Serhy** [comment]
https://github.com/status-im/status-react/pull/5606#issuecomment-421998018 is fixed on iPhoneX. But a regression introduced which has not been present before: in 1-1 chat when contact sends a `/send ETH 0` like request message on the Wallet screen which opens - the password pop-up appears right away. It gives the ability to Sign Transaction with ETH/token amount > than contact has, but the main issue is that we should not show password pop-up right away on `Send transaction` screen with sending transaction in 1-1 chat. 
**To reproduce**:
1) Open Status and log in to account
2) Navigate to 1-1 chat 
3) Send a `/send ETH 1000` message
4) Check whether the `Enter password` pop-up appeared.
**Actual result:** `Enter password` pop-up appears once `Send transaction` screen is opened

TF session: https://app.testfairy.com/projects/4803590-status/builds/8596607/sessions/4401741965/?accessToken=8V9Mkoh3IgW0/KLHaSqJLp4vcXQ

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/8596607/sessions/4401741965/?accessToken=8V9Mkoh3IgW0/KLHaSqJLp4vcXQ">TestFairy: Status 0.9.28 (8472) @2018</a></strong></div></blockquote>
